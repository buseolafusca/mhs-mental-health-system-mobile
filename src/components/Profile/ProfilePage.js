import React from 'react'
import PropTypes from 'prop-types'
import Form from './Form'

const Profile = (props) => (
  <div>
    <Form values={props.profileFields} />
    <footer />
  </div>
)

Profile.propTypes = {
  profileFields: PropTypes.object.isRequired
}

export default Profile
