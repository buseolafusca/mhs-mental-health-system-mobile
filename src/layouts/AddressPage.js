import React from 'react'
import { withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";
import { google } from 'google-maps';

var distance = require('google-distance-matrix');
distance.key('AIzaSyAj0bTPi8uCxzt-Fl_u9t4IHLsGpdGRwfU');
distance.units('imperial');

var NodeGeocoder = require('node-geocoder');


// var options = {
//   provider: 'google',
 
//   // Optional depending on the providers
//   httpAdapter: 'https', // Default
//   apiKey: 'AIzaSyAj0bTPi8uCxzt-Fl_u9t4IHLsGpdGRwfU', // for Mapquest, OpenCage, Google Premier
//   formatter: null         // 'gpx', 'string', ...
// };
 
// var geocoder = NodeGeocoder(options);

// geocoder.geocode('29 champs elys√©e paris', function(err, res) {
//   console.log("res");
//   console.log(res);
//   console.log("err");
//   console.log(err);
// });

// var service = new google.maps.DistanceMatrixService();

// var origin1 = new google.maps.LatLng(55.930385, -3.118425);
// var origin2 = 'Greenwich, England';
// var destinationA = 'Stockholm, Sweden';
// var destinationB = new google.maps.LatLng(50.087692, 14.421150);

// var service = new google.maps.DistanceMatrixService();
// service.getDistanceMatrix(
//   {
//     // origins: [origin1, origin2],
//     // destinations: [destinationA, destinationB],
//     origins: ['San Francisco CA'],
//     destinations: ['New York NY', '41.8337329,-87.7321554'],
//     travelMode: 'DRIVING',
//     // transitOptions: TransitOptions,
//     // drivingOptions: DrivingOptions,
//     // unitSystem: UnitSystem,
//     avoidHighways: Boolean,
//     avoidTolls: Boolean,
//   }, callback);

// function callback(response, status) {
//   console.log("response");
//   console.log(response);
//   console.log("status");
//   console.log(status);

// }


// var bounds = new google.maps.LatLngBounds;
// var markersArray = [];

// var origin1 = {lat: 55.93, lng: -3.118};
// var origin2 = 'Greenwich, England';
// var destinationA = 'Stockholm, Sweden';
// var destinationB = {lat: 50.087, lng: 14.421};

// var destinationIcon = 'https://chart.googleapis.com/chart?' +
//     'chst=d_map_pin_letter&chld=D|FF0000|000000';
// var originIcon = 'https://chart.googleapis.com/chart?' +
//     'chst=d_map_pin_letter&chld=O|FFFF00|000000';
// var map = new google.maps.Map(document.getElementById('map'), {
//   center: {lat: 55.53, lng: 9.4},
//   zoom: 10
// });

// var geocoder = new google.maps.Geocoder;


class AddressPage extends React.Component {

  componentWillMount () {
    var origins = ['San Francisco CA'];
    var destinations = ['New York NY', '41.8337329,-87.7321554'];
    
  
    distance.matrix(origins, destinations, function (err, distances) {      
      console.log("AAA");
      console.log(err);
      console.log(distances);
      
    });

  }

  render () {
    return (
      <div>sss</div>
    )
  }

}

export default AddressPage

