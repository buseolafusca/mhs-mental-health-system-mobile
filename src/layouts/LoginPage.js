import React from 'react';
import { PropTypes } from 'prop-types';
import ErrorText from '../components/ErrorText';
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import * as sessionActions from '../actions/SessionActions';
import { Link } from 'react-router-dom';


class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      Username: '',
      Password: '',
      remember: false,
      error: {}
    };

    this.handleLogin = this.handleLogin.bind(this);
    this.handleCheck = this.handleCheck.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }


  handleLogin() {
    this.setState({
      error: {}
    });

    const { Username, Password, remember } = this.state;
    console.log({'email': Username, 'password': Password})
    this.props.actions.logInUser({'email': Username, 'password': Password});

  }

  componentDidMount() {
    console.log("songchen345@gmail.com")
  }

  handleChange(name, e) {
    this.setState({
      [name]: e.target.value
    });
  }

  handleCheck() {
    this.setState({
      remember: !this.state.remember
    });
  }

  render() {
    let usernameOptions = {
      maxLength: 40,
      placeholder: 'Email',
      type: 'email',
      containerClassName: 'input-container'
    };

    let passwordOptions = {
      maxLength: 40,
      placeholder: 'Password',
      type: 'password',
      containerClassName: 'input-container'
    };

    let formOptions = {
      className: 'login-form'
    };

    let i18n = {
      rememberMe: 'Remember Me',
      submitLabel: 'Sign In'
    };

    if (this.props.username) {
      usernameOptions = Object.assign(usernameOptions, this.props.username);
    }

    if (this.props.password) {
      passwordOptions = Object.assign(passwordOptions, this.props.password);
    }

    if (this.props.form) {
      formOptions = Object.assign(formOptions, this.props.form);
    }

    if (this.props.text) {
      i18n = Object.assign(i18n, this.props.text);
    }

    return (

      <div className='landing-page-container'>
              <NHSHeader />

      <form action="javascript:void(0)" noValidate onSubmit={this.handleLogin} className={formOptions.className}>
        <div className={usernameOptions.containerClassName}>

          <input
            maxLength={usernameOptions.maxLength}
            placeholder={usernameOptions.placeholder}
            autoComplete="off"
            className={usernameOptions.className}
            onChange={e => this.handleChange('Username', e)}
            type={usernameOptions.type}
          />
          <ErrorText errText={this.state.error.email} />
          <span className="glyphicon glyphicon-envelope " />
        </div>
        <div className={passwordOptions.containerClassName}>
          <input
            autoComplete="off"
            className={passwordOptions.className}
            maxLength={40}
            name="Password"
            onChange={e => this.handleChange('Password', e)}
            placeholder={passwordOptions.placeholder}
            type="password"
          />
          <ErrorText errText={this.state.error.password} />
          <span className="glyphicon glyphicon-lock" />
        </div>
        <div style={{
            content: ' ',
            marginRight: -15,
            marginLeft: -15
          }}
        >
          
          <Link to="/register" className="btn btn-link">Register</Link>

          <div className="button-container">
            <button type="submit" className="btnSubmit">{i18n.submitLabel}</button>
          </div>

        </div>
      </form>
        <NHSFooter />

      </div>
    );
  }
}
LoginForm.propTypes = {
  username: PropTypes.object,
  password: PropTypes.object,
  form: PropTypes.object,
  text: PropTypes.object,
  onSubmit: PropTypes.func.isRequired
};

LoginForm.defaultProps = {
  username: {},
  password: {}
};

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(sessionActions, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(LoginForm);

