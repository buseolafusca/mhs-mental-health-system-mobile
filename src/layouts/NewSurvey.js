import React, { Component } from "react";
import * as Survey from "survey-react";
import "survey-react/survey.css";
import logo from "../assets/images/logo.svg";
import "icheck/skins/square/blue.css";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.css";
import "jquery-ui/themes/base/all.css";
import "nouislider/distribute/nouislider.css";
import "select2/dist/css/select2.css";
import "bootstrap-slider/dist/css/bootstrap-slider.css";
import "jquery-bar-rating/dist/themes/css-stars.css";
import $ from "jquery";
import "jquery-ui/ui/widgets/datepicker.js";
import "select2/dist/js/select2.js";
import "jquery-bar-rating";
import * as widgets from "surveyjs-widgets";

import { getQuestionnaire, postAnswers, sendResults } from '../services/BackendService';
import SurveyCreator from "./SurveyCreator";
import "../assets/css/NewSurvey.css";

window["$"] = window["jQuery"] = $;
require("icheck");

Survey.StylesManager.applyTheme("darkblue");

widgets.icheck(Survey, $);
widgets.select2(Survey, $);
widgets.inputmask(Survey);
widgets.jquerybarrating(Survey, $);
widgets.jqueryuidatepicker(Survey, $);
widgets.nouislider(Survey);
widgets.select2tagbox(Survey, $);
widgets.signaturepad(Survey);
widgets.sortablejs(Survey);
widgets.ckeditor(Survey);
widgets.autocomplete(Survey, $);
widgets.bootstrapslider(Survey);

class NewSurvey extends Component {

  constructor(props) {
    super(props);
    this.state = {
      questionnaireId:"5d1a1d16d910160030d04979" ,
      json:
      {
       title: "",
        description: "",
        completedHtml: "",
        pages: [],
        showProgressBar: ""
      }
    };
  }

  /* The following 4 methods are needed by SurveyJS */
  sendResult(){
    console.log("value changed!");
  }

  onValueChanged = (result) => {
    console.log("value changed!");
  }

  sendResultOnPageNext() {
    console.log("sendResultOnPageNext");
  }


  goNextPageAutomatic() {
    console.log("goNextPageAutomatic");
  }


  onComplete = (result) => {
    var finalScore = 0;
    var tableData;
    var i = 1;
    tableData = "<tr><th scope='col'>" + "Question" + "</th><th scope='col'>" + " Answer" + "</th></tr>"
    Object.keys(result.valuesHash).map(function (key) {

      tableData += "<tr>"
      tableData += "<td >" + "Question " + i + "</td>"
      finalScore = finalScore + parseInt(result.valuesHash[key], 10);

      tableData += "<td >" + result.valuesHash[key] + "</td>";
      tableData += "</tr>"
      i++;

    })
    console.log(result);
    console.log(this.model);
    console.log(this.state.questionnaireId);
    console.log(this.model);
    console.log(this.model.data);
    postAnswers(this.model,this.state);
    $("#tbody1").html(tableData);
    document.querySelector('#finalScore').textContent = "Final score is " + finalScore;
    document.querySelector('#jsonSection').textContent = "Result JSON:\n" + JSON.stringify(result.data, null, 3);
  };

  componentWillMount() {
    const { id } = this.props.match.params;
    const url = id;
    this.setState({questionnaireId:id});
    getQuestionnaire(url)
      .then(fetchedData => {
        this.setState({ json: fetchedData.body });
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    Survey.Survey.cssType = "bootstrap";
    this.model = new Survey.Model(this.state.json);
    return (
      <div className="SurveyResult">
        <div className="surveyjs" >
          <Survey.Survey
            model={this.model}
            onComplete={this.onComplete}
            onValueChanged={this.onValueChanged}
          />
          <center>
            <table border="1" width="180" >
              <tbody id="tbody1">

              </tbody>
            </table>
          </center>
          <div id="finalScore"></div>
          <div id="jsonSection"></div>
        </div>

      </div>
    );
    
  }
}

export default NewSurvey;

