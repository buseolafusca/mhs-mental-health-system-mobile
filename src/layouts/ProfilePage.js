import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { userActions } from '../actions/userActions';
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import { getPatientProfile } from '../services/BackendService.js'

class ProfilePage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                firstName: 'ss',
                lastName: '',
                email: '',
                postcode: '',
                telephone: '',
                service: null,
            },
            submitted: false
        };

    }


    componentWillMount () {
        getPatientProfile().then(
            response => {
                // console.log("getPatientProfile")
                console.log(response)
                

                if (response){
                    var newUser = {
                    firstName: response.first_name,
                    lastName: response.last_name,
                    email: response.email,
                    postcode: response.postcode,
                    telephone: 'not provided',
                    service: response.service_id
                }
                  this.setState({ user: newUser})
                }
            }
        )
    }

    render() {
        const { user, submitted } = this.state;
        return (
          <div className='landing-page-container'>
            <NHSHeader />
            <div className="col-md-6 col-md-offset-3">
                <form name="form" >
                    <div className={'form-group' + (submitted && !user.firstName ? ' has-error' : '')}>
                        <label htmlFor="firstName">First Name</label>
                        <input type="text" className="form-control" name="firstName" value={user.firstName} />
                        {submitted && !user.firstName &&
                            <div className="help-block">First Name is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.lastName ? ' has-error' : '')}>
                        <label htmlFor="lastName">Last Name</label>
                        <input type="text" className="form-control" name="lastName" value={user.lastName} />
                        {submitted && !user.lastName &&
                            <div className="help-block">Last Name is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.email ? ' has-error' : '')}>
                        <label htmlFor="email">Email </label>
                        <input type="text" className="form-control" name="email" value={user.email} />
                        {submitted && !user.email &&
                            <div className="help-block">email is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.postcode ? ' has-error' : '')}>
                        <label htmlFor="postcode">Postcode</label>
                        <input type="text" className="form-control" name="postcode" value={user.postcode} />
                        {submitted && !user.postcode &&
                            <div className="help-block">postcode is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.telephone ? ' has-error' : '')}>
                        <label htmlFor="telephone">Telephone</label>
                        <input type="text" className="form-control" name="telephone" value={user.telephone} />
                        {submitted && !user.telephone &&
                            <div className="help-block">telephone is required</div>
                        }
                    </div>
                    <div className={'form-group'}>
                        <label htmlFor="service">Service</label>
                        <input type="text" className="form-control" name="service" value={user.service} />
                    </div>
                        
                    <div className="form-group">
                        <Link to="/landingpage" className="btn btn-link">Back</Link>
                        
                    </div>
                </form>
            </div>
            <NHSFooter />

          </div>
        );
    }
}

const actionCreators = {
    register: userActions.register
}

const connectedProfilePage = connect(null, actionCreators)(ProfilePage);
export { connectedProfilePage as ProfilePage };
