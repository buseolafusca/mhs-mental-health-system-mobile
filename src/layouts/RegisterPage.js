import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { userActions } from '../actions/userActions';
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import { registerUser } from '../services/BackendService.js'
import SelectableTable from '../components/CheckedTable/CheckedTable.js'
import history from '../history'

class RegisterPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                firstName: '',
                lastName: '',
                email: '',
                postcode: '',
                telephone: '',
                password: '',
                repeatedPassword: '',
                service: null,
                postcode: ''
            },
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.myCallback = this.myCallback.bind(this);
    }



    handleChange(event) {
        const { name, value } = event.target;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                [name]: value
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;

        if (user.firstName && user.lastName && user.email && user.password && user.repeatedPassword && user.postcode && user.telephone) {
            if (user.password !== user.repeatedPassword){
                alert("Passwords don't match");
            }
            else{
                registerUser({"email":user.email, "password": user.password, "first_name": user.firstName, "last_name": user.lastName,
            "postcode": user.postcode})
                  .then(fetchedData => {
                    console.log("fetchedData")
                    console.log(fetchedData)
                    alert("Registered successfully");
                    history.push('/login');
                  })
                  .catch(error => {
                    console.error(error);
                  });
            }
        }
    }

    myCallback = (dataFromChild) => {
        console.log("selected service id");
        console.log(dataFromChild);
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                service: dataFromChild
            }
        });
    }

    render() {
        const { registering  } = this.props;
        const { user, submitted } = this.state;
        return (
          <div className='landing-page-container'>
            <NHSHeader />
            <div className="col-md-6 col-md-offset-3">
                <form name="form" onSubmit={this.handleSubmit}>
                    <div className={'form-group' + (submitted && !user.firstName ? ' has-error' : '')}>
                        <label htmlFor="firstName"></label>
                        <input type="text" className="form-control" name="firstName" value={user.firstName} onChange={this.handleChange} placeholder={'First Name'}/>
                        {submitted && !user.firstName &&
                            <div className="help-block">First Name is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.lastName ? ' has-error' : '')}>
                        <label htmlFor="lastName"></label>
                        <input type="text" className="form-control" name="lastName" value={user.lastName} onChange={this.handleChange} placeholder={'Last Name'}/>
                        {submitted && !user.lastName &&
                            <div className="help-block">Last Name is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.email ? ' has-error' : '')}>
                        <label htmlFor="email"></label>
                        <input type="text" className="form-control" name="email" value={user.email} onChange={this.handleChange} placeholder={'Email'}/>
                        {submitted && !user.email &&
                            <div className="help-block">email is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.postcode ? ' has-error' : '')}>
                        <label htmlFor="postcode"></label>
                        <input type="text" className="form-control" name="postcode" value={user.postcode} onChange={this.handleChange} placeholder={'Postcode'}/>
                        {submitted && !user.postcode &&
                            <div className="help-block">postcode is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.telephone ? ' has-error' : '')}>
                        <label htmlFor="telephone"></label>
                        <input type="text" className="form-control" name="telephone" value={user.telephone} onChange={this.handleChange} placeholder={'Telephone'}/>
                        {submitted && !user.telephone &&
                            <div className="help-block">telephone is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.password ? ' has-error' : '')}>
                        <label htmlFor="password"></label>
                        <input type="password" className="form-control" name="password" value={user.password} onChange={this.handleChange} placeholder={'Password'}/>
                        {submitted && !user.password &&
                            <div className="help-block">Password is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.repeatedPassword ? ' has-error' : '')}>
                        <label htmlFor="password"></label>
                        <input type="password" className="form-control" name="repeatedPassword" value={user.repeatedPassword} onChange={this.handleChange} placeholder={'Repeat your password'}/>
                        {submitted && !user.repeatedPassword &&
                            <div className="help-block">Password is required</div>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !user.postcode ? ' has-error' : '')}>
                        <label htmlFor="postcode"></label>
                        <input type="text" className="form-control" name="postcode" value={user.postcode} onChange={this.handleChange} placeholder={'Post Code'}/>
                        {submitted && !user.postcode &&
                            <div className="help-block">Post Code is required</div>
                        }
                    </div>
                    <div>
                        <SelectableTable callbackFromParent={this.myCallback}/>
                    </div>
                        
                 
                    <div className="form-group">
                        <button className="btn btn-primary">Register</button>
                        {registering && 
                            <img src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA==" />
                        }
                        <Link to="/login" className="btn btn-link">Cancel</Link>
                    </div>
                </form>
            </div>
            <NHSFooter />

          </div>
        );
    }
}

function mapState(state) {
    const { registering } = state.registration;
    return { registering };
}

const actionCreators = {
    register: userActions.register
}

const connectedRegisterPage = connect(null, actionCreators)(RegisterPage);
export { connectedRegisterPage as RegisterPage };

