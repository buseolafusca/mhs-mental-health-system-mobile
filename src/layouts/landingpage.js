import React from 'react'
import '../assets/css/LandingPage.css'
import NHSHeader from '../components/NHSHeader.js'
import NHSFooter from '../components/NHSFooter.js'
import { fetchPublishedQuestionnaires } from '../services/BackendService.js'

class Square extends React.Component {
  render () {
    return (
      <a href={this.props.url} className='box' style={{ color: '#fff' }}>{this.props.title}</a>
    )
  }
}

class LandingPage extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      questionnaireList: []
    }
  }

  componentWillMount () {
    fetchPublishedQuestionnaires().then(
      response => {
        this.setState({ questionnaireList: response.questionnaireList })
      }
    )
  }

  render () {
    return (
      <div className='landing-page-container'>
        <NHSHeader />
        <div className='wrapper-container'>
          <div className='wrapper'>
            {this.state.questionnaireList.map((item, key) =>
              <Square url={'/questionnaire/' + item._id} title={item.title} key={key} />
            )}
            <Square url='/referrals' title='Referrals' />
            <Square url='/appointments' title='Appointments' />
            <Square url='/locationpage' title='Resources' />
          </div>
        </div>
        <NHSFooter />
      </div>
    )
  }
}

export default LandingPage
