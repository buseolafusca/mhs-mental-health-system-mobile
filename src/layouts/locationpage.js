import React from 'react';
import { getLocationgivenPostalCode } from './BackendService';
import ReactDOM from 'react-dom';
import NHSHeader from './components/NHSHeader.js'
import NHSFooter from './components/NHSFooter.js'
import ResourcesPage from '../src/resourcespage';
import { BrowserRouter as Router, Link, Route } from 'react-router-dom';
import { Switch } from 'react-router'
import './sass/app.scss';

class LocationPage extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            postcode: '',
            latitude: '',
            longitude: ''
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);

    }

    handleChange(event) {
        this.setState({ postcode: event.target.value });
    }

    handleSubmit(event) {
        //alert('PostCode Submited ' + this.state.postcode);
        getLocationgivenPostalCode(this.state.postcode).then(loc => {
            try {
                this.state.latitude = loc.data.Response.View[0].Result[0].Location.NavigationPosition[0].Latitude;
                this.state.longitude = loc.data.Response.View[0].Result[0].Location.NavigationPosition[0].Longitude;
                var id = this.state.latitude + "," + this.state.longitude;
                this.props.history.push('/resources/' + id);
            } catch (error) {
                console.log("No results Found for that postcode"); //TODO PUT ON A PAGE
            }
        });

        event.preventDefault();

    }


    render() {
        return (
            <Router>
                <Switch>
                    <Route path="/locationpage" exact render={() => {
                        return (
                            <div class="nhsuk-expander-group">
                                <NHSHeader />
                                <form onSubmit={this.handleSubmit}>
                                    <label>
                                        Please enter your postcode:
                                         <input type="text" value={this.state.value} onChange={this.handleChange} />
                                    </label>
                                    <input class="nhsuk-search__submit" type="submit" value="Submit" />
                                    
                                </form>


                                <NHSFooter />
                            </div>

                        );
                    }} />
                    {/* <Route path="/resourcesage" component={ResourcesPage} /> */}
                </Switch>
            </Router>
        );
    }

}

export default LocationPage;