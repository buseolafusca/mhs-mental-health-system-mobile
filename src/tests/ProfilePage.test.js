import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from "react-redux"
import configureMockStore from "redux-mock-store"
import {ProfilePage} from '../layouts/ProfilePage'

const mockStore = configureMockStore();
const store = mockStore({});

describe('Profile page', () => {
  it('should render correctly in mode', () => {
    const component = shallow(<Provider store={store}>
                    <ProfilePage />
                </Provider>)

    expect(component).toMatchSnapshot()

  })

})

