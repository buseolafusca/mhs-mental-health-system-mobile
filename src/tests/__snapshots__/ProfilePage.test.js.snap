// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page should render correctly in mode 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(ProfilePage) />
</ContextProvider>
`;

exports[`Profile page should render correctly in mode 2`] = `
<div
  className="landing-page-container"
>
  <NHSHeader />
  <div
    className="col-md-6 col-md-offset-3"
  >
    <form
      name="form"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="firstName"
        >
          First Name
        </label>
        <input
          className="form-control"
          name="firstName"
          type="text"
          value="ss"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="lastName"
        >
          Last Name
        </label>
        <input
          className="form-control"
          name="lastName"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="email"
        >
          Email 
        </label>
        <input
          className="form-control"
          name="email"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="postcode"
        >
          Postcode
        </label>
        <input
          className="form-control"
          name="postcode"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="telephone"
        >
          Telephone
        </label>
        <input
          className="form-control"
          name="telephone"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="service"
        >
          Service
        </label>
        <input
          className="form-control"
          name="service"
          type="text"
          value={null}
        />
      </div>
      <div
        className="form-group"
      >
        <Link
          className="btn btn-link"
          to="/landingpage"
        >
          Back
        </Link>
      </div>
    </form>
  </div>
  <NHSFooter />
</div>
`;
